app/routes/yara_rules.py

```diff
@@ -562,3 +567,101 @@ def delete_all_inactive_yara_rules():
+    revision_entity = Yara_rule_history.query \
+        .filter_by(yara_rule_id=yara_rule_id) \
+        .filter_by(revision=revision) \
+        .first()
+
+    if not current_entity or not revision_entity:
+        abort(404)
+
+    if not current_user.admin and current_entity.owner_user_id != current_user.id:
+        abort(403)
+
+    revision_dict = revision_entity.to_dict()
+    yara_revision_dict = revision_dict["rule_json"]
+
+    temp_sig_id = current_entity.eventid
+    if not current_entity.category == yara_revision_dict['category']:
+        temp_sig_id = CfgCategoryRangeMapping.get_next_category_eventid(yara_revision_dict['category'])
+
+    db.session.add(yara_rule.Yara_rule_history(date_created=datetime.datetime.now(), revision=current_entity.revision,
+                                               rule_json=json.dumps(current_entity.to_revision_dict()),
+                                               user_id=current_user.id,
+                                               yara_rule_id=current_entity.id,
+                                               state=current_entity.state))
+
+    current_entity = yara_rule.Yara_rule(
+        state=yara_revision_dict['state'],
+        name=yara_revision_dict['name'],
+        description=yara_revision_dict['description'],
+        references=yara_revision_dict['references'],
+        category=yara_revision_dict['category'],
+        condition=re.sub('^condition:\\n\\t', '', yara_revision_dict['condition']),
+        strings=re.sub('^strings:\\n\\t', '', yara_revision_dict['strings']),
+        eventid=temp_sig_id,
+        id=yara_rule_id,
+        creation_date=yara_revision_dict['creation_date'],
+        modified_user_id=current_user.id,
+        last_revision_date=datetime.datetime.now(),
+        owner_user_id=yara_revision_dict['owner_user']['id'] if yara_revision_dict['owner_user'] else None,
+        revision=current_entity.revision + 1,
+        imports=yara_revision_dict['imports'],
+        active=yara_revision_dict['active'],
+        mitre_techniques=yara_revision_dict['mitre_techniques'],
+        mitre_tactics=yara_revision_dict['mitre_tactics'],
+        files=yara_revision_dict['files'],
+    )
+
+    mitre_techniques = Cfg_settings.get_setting("MITRE_TECHNIQUES").split(",")
+    matches = [technique for technique in current_entity.mitre_techniques if technique not in mitre_techniques]
+    if matches:
+        raise (Exception(
+            "The following techniques were not found in the configuration: %s. Check 'MITRE_TECHNIQUES' on the settings page" % (
+                matches)))
+
+    mitre_tactics = Cfg_settings.get_setting("MITRE_TACTICS").split(",")
+    matches = [tactic for tactic in current_entity.mitre_tactics if tactic not in mitre_tactics]
+    if matches:
+        raise (Exception(
+            "The following tactics were not found in the configuration: %s. Check 'MITRE_TACTICS' on the settings page" % (
+                matches)))
```

Comment:
Copy/Pasted this mitre logic from the `update` method... let me know if it's not needed here